modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/LimitNode.java

```diff
@@ -89,19 +93,30 @@ public void push(RowT row) throws Exception {
         if (waiting == NOT_WAITING) {
             return;
         }
+
         ++rowsProcessed;
 
         --waiting;
 
         if (rowsProcessed > offset) {
-            if (fetchNode == null || rowsProcessed <= fetch + offset) {
+            if (fetchUndefined || rowsProcessed <= fetch + offset) {
```

Comment:
looks like `!fetchNone()`

Reply:
it`s not true, just change it and run : ItLimitOffsetTest#testLimitOffset all this checks have been wrote by blood)